virtualization_arm64.m

```diff
@@ -24,16 +24,21 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
  */
 void *newVZMacAuxiliaryStorageWithCreating(const char *storagePath, void *hardwareModel, void **error)
 {
-    VZMacAuxiliaryStorage *auxiliaryStorage;
-    @autoreleasepool {
-        NSString *storagePathNSString = [NSString stringWithUTF8String:storagePath];
-        NSURL *storageURL = [NSURL fileURLWithPath:storagePathNSString];
-        auxiliaryStorage = [[VZMacAuxiliaryStorage alloc] initCreatingStorageAtURL:storageURL
-                                                                     hardwareModel:(VZMacHardwareModel *)hardwareModel
-                                                                           options:VZMacAuxiliaryStorageInitializationOptionAllowOverwrite
-                                                                             error:(NSError *_Nullable *_Nullable)error];
+    if (@available(macOS 12, *)) {
+        VZMacAuxiliaryStorage *auxiliaryStorage;
+        @autoreleasepool {
+            NSString *storagePathNSString = [NSString stringWithUTF8String:storagePath];
+            NSURL *storageURL = [NSURL fileURLWithPath:storagePathNSString];
+            auxiliaryStorage = [[VZMacAuxiliaryStorage alloc] initCreatingStorageAtURL:storageURL
+                                                                         hardwareModel:(VZMacHardwareModel *)hardwareModel
+                                                                               options:VZMacAuxiliaryStorageInitializationOptionAllowOverwrite
+                                                                                 error:(NSError *_Nullable *_Nullable)error];
+        }
+        return auxiliaryStorage;
+    } else {
+        [[NSException exceptionWithName:@"UnhandledException" reason:@"bug" userInfo:nil] raise];
+        return nil;
     }
```

Comment:
Could you fix like this

```objective-c
if (@available(macOS 12, *)) {
  // code
  return auxiliaryStorage; 
}
[[NSException exceptionWithName:@"UnhandledException" reason:@"bug" userInfo:nil] raise];
// should not return here because unreachable code
```

Reply:
Without an explicit `return`, there's a compiler warning:
```
warning: non-void function does not return a value in all control paths [-Wreturn-type]
```
Another option is to use `__builtin_unreachable();`:
```
if (@available(macOS 12, *)) {
  // code
  return auxiliaryStorage; 
}
[[NSException exceptionWithName:@"UnhandledException" reason:@"bug" userInfo:nil] raise];
__builtin_unreachable();
```

Reply:
Good. It's better to make it in macro 👍

Reply:
Done.